packages/core/src/elements-context.ts

```diff
@@ -0,0 +1,85 @@
+import { InternalSchemaElement } from './typeschema/types';
+import { getPathDifference } from './utils';
+
+export type ElementsContextType = {
```

Comment:
Can we add some comments to this?  This type seems important.

Reply:
Yes, that's a great idea. The usage of ElementsContext has slowly expanded and definitely deserves some documentation TLC.